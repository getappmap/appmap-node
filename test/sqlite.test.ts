import { integrationTest, readAppmap, runAppmapNode } from "./helpers";

integrationTest("mapping Sqlite tests", () => {
  expect(runAppmapNode("index.js").status).toBe(0);
  expect(readAppmap()).toMatchSnapshot();
});
