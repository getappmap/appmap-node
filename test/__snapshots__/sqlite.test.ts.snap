// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapping Sqlite tests 1`] = `
{
  "classMap": [
    {
      "children": [
        {
          "children": [
            {
              "location": "index.js:13",
              "name": "main",
              "static": true,
              "type": "function",
            },
            {
              "location": "index.js:5",
              "name": "promisify",
              "static": true,
              "type": "function",
            },
          ],
          "name": "index",
          "type": "class",
        },
      ],
      "name": "sqlite-appmap-node-test",
      "type": "package",
    },
  ],
  "events": [
    {
      "defined_class": "index",
      "event": "call",
      "id": 1,
      "lineno": 13,
      "method_id": "main",
      "parameters": [],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 2,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.run'"",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 3,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.run'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 4,
      "parent_id": 2,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 2,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 6,
      "parent_id": 3,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 7,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.all'"",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 8,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.all'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 9,
      "parent_id": 7,
      "return_value": {
        "class": "Promise<Array>",
        "object_id": 4,
        "value": "Promise { [ { "'Database.all'": 'Database.all' } ] }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 10,
      "parent_id": 8,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 11,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.get'"",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 12,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.get'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 13,
      "parent_id": 11,
      "return_value": {
        "class": "Promise<Object>",
        "object_id": 5,
        "value": "Promise { { "'Database.get'": 'Database.get' } }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 14,
      "parent_id": 12,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 15,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.each'"",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 16,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.each'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 17,
      "parent_id": 15,
      "return_value": {
        "class": "Promise<Object>",
        "object_id": 6,
        "value": "Promise { { "'Database.each'": 'Database.each' } }",
      },
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 19,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: exec]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.exec'"",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 20,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.exec'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 21,
      "parent_id": 19,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 7,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 22,
      "parent_id": 20,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 23,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.each with row callback'"",
        },
        {
          "class": "Function",
          "value": "[Function (anonymous)]",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 24,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.each with row callback'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 25,
      "parent_id": 23,
      "return_value": {
        "class": "Promise<Number>",
        "object_id": 8,
        "value": "Promise { 1 }",
      },
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 27,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.run with params', ? AS a, ? AS b"",
        },
        {
          "class": "Number",
          "value": "1",
        },
        {
          "class": "Number",
          "value": "2",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 28,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.run with params', ? AS a, ? AS b",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 29,
      "parent_id": 27,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 9,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 30,
      "parent_id": 28,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 31,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function (anonymous)]",
        },
        {
          "class": "Database",
          "name": "thisArg",
          "object_id": 1,
          "value": "Database {}",
        },
        {
          "class": "String",
          "value": ""SELECT 'Database.run with object param', $foo, $bar"",
        },
        {
          "class": "Object",
          "object_id": 10,
          "properties": [
            {
              "class": "Number",
              "name": "$foo",
            },
            {
              "class": "String",
              "name": "$bar",
            },
          ],
          "value": "{ '$foo': 123, '$bar': 'abc' }",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 32,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.run with object param', $foo, $bar",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 33,
      "parent_id": 31,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 11,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 34,
      "parent_id": 32,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 35,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: run]",
        },
        {
          "class": "Statement",
          "name": "thisArg",
          "object_id": 12,
          "value": "Statement {}",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 36,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.run'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 37,
      "parent_id": 35,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 13,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 38,
      "parent_id": 36,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 39,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: all]",
        },
        {
          "class": "Statement",
          "name": "thisArg",
          "object_id": 14,
          "value": "Statement {}",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 40,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.all'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 41,
      "parent_id": 39,
      "return_value": {
        "class": "Promise<Array>",
        "object_id": 15,
        "value": "Promise { [ { "'Statement.all'": 'Statement.all' } ] }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 42,
      "parent_id": 40,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 43,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: get]",
        },
        {
          "class": "Statement",
          "name": "thisArg",
          "object_id": 16,
          "value": "Statement {}",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 44,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.get'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 45,
      "parent_id": 43,
      "return_value": {
        "class": "Promise<Object>",
        "object_id": 17,
        "value": "Promise { { "'Statement.get'": 'Statement.get' } }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 46,
      "parent_id": 44,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 47,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: each]",
        },
        {
          "class": "Statement",
          "name": "thisArg",
          "object_id": 18,
          "value": "Statement {}",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 48,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.each'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 49,
      "parent_id": 47,
      "return_value": {
        "class": "Promise<Object>",
        "object_id": 19,
        "value": "Promise { { "'Statement.each'": 'Statement.each' } }",
      },
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 51,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: run]",
        },
        {
          "class": "Statement",
          "name": "thisArg",
          "object_id": 20,
          "value": "Statement {}",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 52,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.run - prepare once run twice'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 53,
      "parent_id": 51,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 21,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 54,
      "parent_id": 52,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 55,
      "lineno": 5,
      "method_id": "promisify",
      "parameters": [
        {
          "class": "Function",
          "name": "method",
          "value": "[Function: run]",
        },
        {
          "class": "Statement",
          "name": "thisArg",
          "object_id": 20,
          "value": "Statement { lastID: 0, changes: 0 }",
        },
      ],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 56,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.run - prepare once run twice'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 57,
      "parent_id": 55,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 22,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 58,
      "parent_id": 56,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 59,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Database.run without a completion callback'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 60,
      "parent_id": 59,
      "thread_id": 0,
    },
    {
      "event": "call",
      "id": 61,
      "sql_query": {
        "database_type": "sqlite",
        "sql": "SELECT 'Statement.run without a completion callback'",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 62,
      "parent_id": 61,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 5,
      "parent_id": 1,
      "return_value": {
        "class": "Promise<undefined>",
        "object_id": 3,
        "value": "Promise { undefined }",
      },
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 18,
      "parent_id": 16,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 26,
      "parent_id": 24,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 50,
      "parent_id": 48,
      "thread_id": 0,
    },
  ],
  "metadata": {
    "app": "sqlite-appmap-node-test",
    "client": {
      "name": "appmap-node",
      "url": "https://github.com/getappmap/appmap-node",
      "version": "test node-appmap version",
    },
    "language": {
      "engine": "Node.js",
      "name": "javascript",
      "version": "test node version",
    },
    "name": "test process recording",
    "recorder": {
      "name": "process",
      "type": "process",
    },
  },
  "version": "1.12",
}
`;
